perception/image_projection_based_fusion/src/fusion_node.cpp

```diff
@@ -157,67 +143,159 @@ void FusionNode<Msg, Obj>::preprocess(Msg & ouput_msg __attribute__((unused)))
 }
 
 template <class Msg, class Obj>
-void FusionNode<Msg, Obj>::fusionCallback(
-  typename Msg::ConstSharedPtr input_msg, DetectedObjectsWithFeature::ConstSharedPtr input_roi0_msg,
-  DetectedObjectsWithFeature::ConstSharedPtr input_roi1_msg,
-  DetectedObjectsWithFeature::ConstSharedPtr input_roi2_msg,
-  DetectedObjectsWithFeature::ConstSharedPtr input_roi3_msg,
-  DetectedObjectsWithFeature::ConstSharedPtr input_roi4_msg,
-  DetectedObjectsWithFeature::ConstSharedPtr input_roi5_msg,
-  DetectedObjectsWithFeature::ConstSharedPtr input_roi6_msg,
-  DetectedObjectsWithFeature::ConstSharedPtr input_roi7_msg)
+void FusionNode<Msg, Obj>::subCallback(const typename Msg::ConstSharedPtr input_msg)
 {
-  Msg output_msg = *input_msg;
-
-  preprocess(output_msg);
-
-  for (std::size_t image_id = 0; image_id < rois_subs_.size(); ++image_id) {
-    DetectedObjectsWithFeature::ConstSharedPtr input_roi_msg;
-    switch (image_id) {
-      case 0:
-        input_roi_msg = input_roi0_msg;
-        break;
-      case 1:
-        input_roi_msg = input_roi1_msg;
-        break;
-      case 2:
-        input_roi_msg = input_roi2_msg;
-        break;
-      case 3:
-        input_roi_msg = input_roi3_msg;
-        break;
-      case 4:
-        input_roi_msg = input_roi4_msg;
-        break;
-      case 5:
-        input_roi_msg = input_roi5_msg;
-        break;
-      case 6:
-        input_roi_msg = input_roi6_msg;
-        break;
-      case 7:
-        input_roi_msg = input_roi7_msg;
-        break;
-      default:
-        RCLCPP_ERROR(this->get_logger(), "invalid image id. id is %zu", image_id);
-        return;
-    }
+  std::lock_guard<std::mutex> lock(mutex_);
+  auto period = std::chrono::duration_cast<std::chrono::nanoseconds>(
+    std::chrono::duration<double>(timeout_sec_));
+  try {
+    setPeriod(period.count());
+  } catch (rclcpp::exceptions::RCLError & ex) {
+    RCLCPP_WARN_THROTTLE(get_logger(), *get_clock(), 5000, "%s", ex.what());
+  }
+  timer_->reset();
+
+  stop_watch_ptr_->toc("processing_time", true);
+
+  typename Msg::SharedPtr output_msg = std::make_shared<Msg>(*input_msg);
 
-    if (camera_info_map_.find(image_id) == camera_info_map_.end()) {
-      RCLCPP_WARN(this->get_logger(), "no camera info. id is %zu", image_id);
+  preprocess(*output_msg);
+
+  int64_t nanosec =
```

Comment:
It is  easier to understand to change to `timestamp_nsec`

Reply:
fixed in https://github.com/autowarefoundation/autoware.universe/pull/2175/files#diff-d7a2406737247406518fdab67366a31dad5e5f8ddf94f13e661e4c22d8ee0633